tsdb/store.go

```diff
@@ -33,6 +33,9 @@ var (
 	ErrStoreClosed = fmt.Errorf("store is closed")
 	// ErrShardDeletion is returned when trying to create a shard that is being deleted
 	ErrShardDeletion = errors.New("shard is being deleted")
+	// ErrMultipleIndexTypes is returned when trying to do deletes on a database with
+	// multiple index types.
+	ErrMultipleIndexTypes = errors.New("cannot delete with multiple index types")
```

Comment:
Can we expand on this a bit? I think this is going to surprise some users, so we should be quick to help resolve the situation for them. Maybe something like:

```
cannot delete data. DB contains shards using both inmem and tsi1 indexes. Please convert all shards to use the same index type to delete data. 
```

Or something like that...